app/services/flow/proceeding_loop.rb

```diff
@@ -59,7 +72,7 @@ def date_confirmation_required
     alias_method :date_confirmation_required?, :date_confirmation_required
 
     def at_end_of_loop?
-      final_proceeding_in_loop.id == @application.provider_step_params["id"] && %w[delegated_functions confirm_delegated_functions_date].include?(@application.provider_step)
+      final_proceeding_in_loop.id == @application.provider_step_params["id"] && controllers[-2..].include?(@application.provider_step)
```

Comment:
I found this a little opaque, possibly worth a comment/explaining that the order of the controller loops matters?